dev-libs/wlroots/wlroots-9999.ebuild

```diff
@@ -0,0 +1,118 @@
+# Copyright 1999-2018 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+
+if [[ ${PV} == 9999 ]] ; then
+	EGIT_REPO_URI="https://github.com/swaywm/wlroots.git"
+	inherit git-r3
+else
+	SRC_URI="https://github.com/swaywm/wlroots/archive/${PV}.tar.gz -> ${P}.tar.gz"
+	KEYWORDS="~amd64 ~x86"
+fi
+
+inherit fcaps meson
+
+DESCRIPTION="Pluggable, composable, unopinionated modules for building a Wayland compositor"
+HOMEPAGE="https://github.com/swaywm/wlroots"
+LICENSE="MIT"
+SLOT="0"
+IUSE="elogind examples filecaps png rootston systemd x11-backend xcb-icccm xcb-xkb xwayland"
+REQUIRED_USE="?? ( elogind systemd )"
+
+# An example requires >=ffmpeg-4.0 (actually masked) or equivalent libav
+# (actually upstream is behind ffmpeg-4.0). So the ffmpeg use flag is actually
+# not present and the related example (only one) will not be build. This is a
+# reminder to add the 'ffmpeg' use flag when ffmpeg-4.0 will be unmasked.
+# examples? ( ffmpeg? ( >=media-video/ffmpeg-4.0 ) )
+RDEPEND=">=dev-libs/libinput-1.7.0:0=
+	>=dev-libs/wayland-1.16.0
+	>=dev-libs/wayland-protocols-1.15
+	media-libs/mesa[egl,gles2,gbm]
+	virtual/libudev
+	x11-libs/libdrm
+	x11-libs/libxkbcommon
+	x11-libs/pixman
+	filecaps? ( sys-libs/libcap )
+	elogind? ( >=sys-auth/elogind-237 )
+	examples? (
+		png? ( media-libs/libpng:* )
```

Comment:
libpng slots 1.2 and 1.5 don't provide headers, so I honestly doubt they are going to work here.